cheroot/__main__.py

```diff
@@ -0,0 +1,134 @@
+"""Command line tool for starting a Cheroot instance.
+
+Basic usage::
+
+    # Start a server on 127.0.0.1:8000 with the default settings
+    # for the WSGI app myapp/wsgi.py:application()
+    cheroot myapp.wsgi
+
+    # Start a server on 0.0.0.0:9000 with 8 threads
+    # for the WSGI app myapp/wsgi.py:main_app()
+    cheroot myapp.wsgi:main_app --bind 0.0.0.0:9000 --threads 8
+"""
+
+import argparse
+from importlib import import_module
+import os
+import re
+import sys
+
+from .wsgi import Server
+
+RE_HOST_PORT = r'^(\S+)(?::)([0-9]+)$'
+
+
+def wsgi_app_path(string):
+    """Convert WSGI app string to a tuple."""
+    if ':' in string:
+        mod_path, app_path = string.split(':', 1)
+    else:
+        mod_path, app_path = string, 'application'
+
+    module = import_module(mod_path)
+    app = eval(app_path, vars(module))
+
+    if app is None:
+        raise NameError(
+            'Failed to find application object: {}'.format(app_path))
+    elif not callable(app):
+        raise TypeError('Application must be a callable object')
+
+    return app
+
+
+def wsgi_bind_addr(string):
+    """Convert bind address string to a format accepted by Server()."""
+    # try and match for an IP/hostname and port
+    match = re.match(RE_HOST_PORT, string)
+    if match:
+        addr, port = match.groups()
+        port = int(port)
+        return addr, port
+
+    # else, assume a UNIX socket path
+    return string
+
+
+def main(args=None):
+    """Run cheroot from the CLI."""
```

Comment:
This docstring is misleading. This function is already being run invoked from CLI, it doesn't invoke a new one.